benchopt/benchmark.py

```diff
@@ -295,37 +385,64 @@ def install_all_requirements(self, include_solvers, include_datasets,
             )
         print(' done')
 
-    def validate_dataset_patterns(self, dataset_patterns):
-        "Check that all provided patterns match at least one dataset"
-
-        # List all dataset strings.
-        all_datasets = _list_all_parametrized_names(*self.get_datasets())
-        all_datasets += ["all"]
-
-        _validate_patterns(all_datasets, dataset_patterns, name_type='dataset')
-
-    def validate_solver_patterns(self, solver_patterns):
-        "Check that all provided patterns match at least one solver"
-
-        # List all dataset strings.
-        all_solvers = _list_all_parametrized_names(*self.get_solvers())
-        all_solvers += ["all"]
-
-        _validate_patterns(all_solvers, solver_patterns, name_type='solver')
-
-    def validate_objective_filters(self, objective_params):
-        "Check that all objective filters match at least one objective setup"
+    def get_all_runs(self, solver_names=None, forced_solvers=None,
+                     dataset_names=None, objective_filters=None, output=None):
+        """Generator with all combinations to run for the benchmark.
 
-        # List all choices of objective parameters
-        all_objectives = _list_all_parametrized_names(
-            self.get_benchmark_objective()
-        )
-
-        _validate_patterns(
-            all_objectives,
-            objective_params,
-            name_type="objective"
-        )
+        Parameters
+        ----------
+        solver_names : list | None
+            List of solvers to include in the benchmark. If None
+            all solvers available are run.
+        forced_solvers : list | None
+            List of solvers to include in the benchmark and for
+            which one forces recomputation.
+        dataset_names : list | None
+            List of datasets to include. If None all available
+            datasets are used.
+        objective_filters : list | None
+            Filters to select specific objective parameters. If None,
+            all objective parameters are tested
+        output : TerminalOutput or None
+            Object to format string to display the progress of the solver.
```

Comment:
I don't understand the meaning of "object to format string to display"

Reply:
Is this better?
```suggestion
            Object to manage the output in the terminal.
```

Reply:
yes perfect